cfme/tests/intelligence/reports/test_reports.py

```diff
@@ -239,3 +274,55 @@ def test_report_edit_secondary_display_filter(appliance, filter_report, soft_ass
         view.report_info.secondary_filter.read() == secondary_filter,
         "Secondary Filter did not match.",
     )
+
+
+@test_requirements.report
+@pytest.mark.tier(1)
+@pytest.mark.meta(server_roles="+notifier", automates=[1677839])
+@pytest.mark.provider([InfraProvider], selector=ONE_PER_CATEGORY)
+def test_send_text_custom_report_with_long_condition(
+    appliance, setup_provider, smtp_test, soft_assert, request, generate_report
+):
+    """
+    Polarion:
+        assignee: pvala
+        casecomponent: Reporting
+        caseimportance: medium
+        initialEstimate: 1/3h
+        setup:
+            1. Create a report containing 1 or 2 columns
+                and add a report filter with a long condition.(Refer BZ for more detail)
+            2. Create a schedule for the report and check send_txt.
+        testSteps:
+            1. Queue the schedule and monitor evm log.
+        expectedResults:
+            1. There should be no error in the log and report must be sent successfully.
+
+    Bugzilla:
+        1677839
+    """
+    report = generate_report(LONG_CONDITION_DATA)
+    data = {
+        "timer": {"hour": "12", "minute": "10"},
+        "email": {"to_emails": "test@example.com"},
+        "email_options": {"send_if_empty": True, "send_txt": True},
+    }
+    schedule = report.create_schedule(**data)
+
+    # prepare LogValidator
+    pattern = ".*negative argument.*"
+    log = LogValidator("/var/www/miq/vmdb/log/evm.log", matched_patterns=[pattern])
+
+    log.start_monitoring()
+    schedule.queue()
+
+    # assert that the mail was sent
+    wait_for(
+        lambda: len(smtp_test.get_emails(to_address=data["email"]["to_emails"])) == 1,
+        timeout=200,
+        delay=5,
+        msg="Mail was not sent. Some error occured.",
+    )
+
+    # assert that the pattern was not found in the logs
+    soft_assert(not log.validate(), "Found error message in the logs.")
```

Comment:
@valaparthvi You can use the `failure_pattern` keyword if you're trying to assert that a pattern **shouldn't** be found in the logs. https://github.com/ManageIQ/integration_tests/blob/861426751dd237366a91410fa0b76170d0351eb2/cfme/utils/log_validator.py#L57

Reply:
Oh, that is how `failure_pattern` works! I had used it earlier, it wasn't working as I'd hoped, or at least that's what I thought. Will change that, thanks John! :)